test/structure/structure_region/test_capping.py

```diff
@@ -24,35 +25,184 @@ def test_capping_with_residue_terminals(helpers):
+            test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+            capping.capping_with_residue_terminals(test_region, nterm_cap=nterm_cap, cterm_cap=cterm_cap)
+   
+            unique_names = set()
+            for atom in test_region.atoms:
+                unique_names.add(atom.name)
+        
+            assert len(test_region.atoms) == len(unique_names)
+
+
+
+def test_all_cap_types_have_realistic_bond_distances():
+    """Checking that all the cap types have realistic bond distances with their link Residue()'s and Atom()'s."""
+    test_stru = sp.get_structure(f"{DATA_DIR}tri_alanine.pdb")
+    sele = stru_sele.select_stru(
+        test_stru, "resi 2")
+
+    cap_types:List[str] = list(capping.SUPPORTED_CAPS.keys())
+
+    for nterm_cap in cap_types:
+        for cterm_cap in cap_types:
+            test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+            capping.capping_with_residue_terminals(test_region, nterm_cap=nterm_cap, cterm_cap=cterm_cap)
+         
+            for atom in test_region.atoms:
+                if not isinstance(atom.parent, capping.ResidueCap):
+                    continue
+                link_distance:float=atom.parent.link_atom.distance_to(atom.parent.socket_atom)
+                assert link_distance >= 1.0 and link_distance <= 1.50
+
+
+
+def test_correct_num_caps():
+    """Testing that the correct number of caps are made."""
+
+    test_stru = sp.get_structure(f"{DATA_DIR}tri_alanine.pdb")
+    sele = stru_sele.select_stru(
+        test_stru, "resi 1+2+3")
+
+    test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+
+    caps = set()
+
+    capping.capping_with_residue_terminals(test_region)
+
+    for atom in test_region.atoms:
+        if not isinstance(atom.parent, capping.ResidueCap):
+            continue
+        caps.add(atom.parent)
+
+    assert len(caps) == 0
+
+
+    sele = stru_sele.select_stru(
+        test_stru, "resi 1")
+
+    test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+
+    caps = set()
+
+    capping.capping_with_residue_terminals(test_region)
+
+    for atom in test_region.atoms:
+        if not isinstance(atom.parent, capping.ResidueCap):
+            continue
+        caps.add(atom.parent)
+
+    assert len(caps) == 1 
+
+
+    sele = stru_sele.select_stru(
+        test_stru, "resi 3")
+
+    test_region = stru_regi.StructureRegion(atoms=sele.atoms)
+
+    caps = set()
+
     capping.capping_with_residue_terminals(test_region)
-    # test
-    atoms = test_region.atoms
-    test_file = f"{WORK_DIR}test_capping_1.xyz"
-    answer_file = f"{DATA_DIR}answer_capping_1.xyz"
-    lines = [
-        str(len(atoms)),
-        "",]
-    for at in atoms:
-        lines.append(f"{at.element} {at.coord[0]} {at.coord[1]} {at.coord[2]}")
-    fs.write_lines(test_file, lines)
-    assert helpers.equiv_files(test_file, answer_file, consider_order=False)
-
-    fs.safe_rm(test_file)
-
-def test_residue_cap_deepcopy():
```

Comment:
please also keep this test

Reply:
Sure

Reply:
I have tried a few ways but i don't think there is a way to do this test without changing it a lot in the current scheme. thoughts?

Reply:
[after meeting]
Just need to create a ResidueCap() differently and test the deepcopy()

Reply:
Added this back in! Seems to be working ok now.